.github/workflows/test.yml

```diff
@@ -4,7 +4,7 @@ on:
   push:
 
 env:
-  CURRENT_STABLE_CHANNEL: nixpkgs-22.11-darwin
+  CURRENT_STABLE_CHANNEL: nixpkgs-23.05-darwin
```

Comment:
With regard to support for releases, having only 1 option for a stable release while it was only recently created feels a little early to me.
That said, given that the conversion is rather impactful and only for documentation, would it be possible to make a 22.11 build possible if we set the default for `documentation.enable = false` for systems with an old nixpkgs?

Reply:
Actually evaluation works as is on 22.11 so this should be all that's needed. Would be nice if something similar is still possible with the next step.

```patch
diff --git a/modules/documentation/default.nix b/modules/documentation/default.nix
index 15ccfb2..3f8179d 100644
--- a/modules/documentation/default.nix
+++ b/modules/documentation/default.nix
@@ -99,12 +99,14 @@ in
     programs.man.enable = cfg.man.enable;
     programs.info.enable = cfg.info.enable;

-    environment.systemPackages = mkMerge [
+    environment.systemPackages = mkIf (lib.versionAtLeast lib.version "23.05") (mkMerge [
       (mkIf cfg.man.enable [ manual.manpages ])
       (mkIf cfg.doc.enable [ manual.manualHTML helpScript ])
-    ];
+    ]);

     system.build.manual = manual;

+    warnings = optional (lib.versionOlder lib.version "22.11") "documentation.`enable` now requires 23.05 to build documentation for `configuration.nix`, only manpages for system packages will be enabled.";
+
   };
 }
```

Reply:
Do we want to support stable releases that upstream nixpkgs don't consider supported, in general? It's possible to pin an older version of nix-darwin and search.nixos.org lists 22.11 as deprecated already (although [the release notes](https://nixos.org/manual/nixos/stable/release-notes.html#sec-release-22.11) imply support will stop at the end of June, which I guess would give it another week from now). My impression is that people following the stable channels of nixpkgs/NixOS are expected to ride the upgrade train pretty proactively (which admittedly is one of the reasons I've never seen the point).

It'd be easy to disable documentation on 22.11, though, so I'm happy to make this change if you'd prefer it; just wondering about what the general approach is here.

Reply:
Yeah, I personally find a couple of months rather short. Given that's the policy upstream there's no point to bend over backwards, but if we can make it work without docs using a few lines let's do that.

Reply:
Okay, I've handled this in the newest push and confirmed that things still seem to build fine on 22.11. I added stub packages that replace the manual with notices explaining they require 23.05; since `documentation.enable` is on by default it seemed excessive to print a warning on every activation even if people aren't using the manual at all, but if you'd prefer to do the simpler thing I can rip that out.

Reply:
I think explicit warnings are fine in this case, but the stub is also good.

ps. After thinking about it a bit more. With a nixpkgs release nothing really breaks, it will just stop receiving updates until they explicitly update. So handling a bit more compatibility here is kind of the tradeoff for not having releases (which seems a bit overkill given the low rate of change).